main.go

```diff
@@ -32,3 +40,65 @@ func main() {
 		panic(err.Error())
 	}
 }
+
+func getIPAMClient(ctx context.Context, root settings.Source) (*ipamfetcher.Client, error) {
+	ipsPrefixedEnv := &settings.PrefixSource{
+		Source: root,
+		Prefix: []string{"IPS"},
```

Comment:
open to opinions on a better way to configure multiple components with similar settings

Reply:
There's nothing inherently wrong with using the `PrefixSource`. It's there to help with complex cases. Another option might be to use a single top-level config that embeds the repeated config a number of times. For example:

```golang
type RouteConfig struct {
  Endpoint string `description:"Route for the client to use."`
  Limit int `description:"Max concurrent calls"`
}

type SubnetsConfig RouteConfig
func(*SubnetsConfig) Name() string { return "subnets" }
type CustomersConfig RouteConfig
func(*CustomersConfig) Name() string { return "customers" }
type IPSConfig RouteConfig
func(*IPSConfig) Name() string { return "ips" }

type IPAMConfig struct {
  Subnets *SubnetsConfig
  Customers *CustomersConfig
  IPS *IPSConfig
}
```

That would express the same hierarchy using the same names as you do here but with structs instead of prefixes.